spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSqlExtensionsAstBuilder.scala

```diff
@@ -267,6 +286,12 @@ class IcebergSqlExtensionsAstBuilder(delegate: ParserInterface) extends IcebergS
   private def typedVisit[T](ctx: ParseTree): T = {
     ctx.accept(this).asInstanceOf[T]
   }
+
+  private def timeRetain(unit: String, duration: Long): Long = unit.toUpperCase(Locale.ENGLISH) match {
+    case "MONTHS" => 30 * TimeUnit.DAYS.toMillis(duration)
+    case "DAYS" | "HOURS" | "MINUTES" => TimeUnit.valueOf(unit.toUpperCase(Locale.ENGLISH)).toMillis(duration)
```

Comment:
do we still need to use match statement? The parser will only allow these values, so we should be able to directly parse it to TimeUnit.

Reply:
Done.